### ISSUE TITLE
Query failure: Invalid value 'Invalid date': expected a DateTime

### ISSUE THREAD

**Issue by julianespinel** [metabase/metabase#8287]
### Bug

`Query failure: Invalid value 'Invalid date': expected a DateTime`

- Your browser and the version: Chrome 67
- Your operating system: AWS 64bit Amazon Linux/2.7.2
- Your databases: Postgres
- Metabase version: 0.30
- Metabase hosting environment: Elastic Beanstalk
- Metabase internal database: Postgres

- *Repeatable steps to reproduce the issue*

After executing multiple queries with grouping and dates, we see the following errors in logs:

```
08-10 22:12:05 WARN metabase.query-processor :: Query failure: Invalid value 'Invalid date': expected a DateTime.
["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:217)"
 "query_processor$assert_query_status_successful.invoke(query_processor.clj:210)"
 "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:250)"
 "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:243)"
 "query_processor$fn__36878$process_query_and_save_execution_BANG___36883$fn__36884.invoke(query_processor.clj:289)"
 "query_processor$fn__36878$process_query_and_save_execution_BANG___36883.invoke(query_processor.clj:275)"
 "query_processor$fn__36902$process_query_and_save_with_max_BANG___36907$fn__36908.invoke(query_processor.clj:310)"
 "query_processor$fn__36902$process_query_and_save_with_max_BANG___36907.invoke(query_processor.clj:306)"
 "api.dataset$fn__43354$fn__43357.invoke(dataset.clj:45)"
 "api.common$fn__20621$invoke_thunk_with_keepalive__20626$fn__20627$fn__20628.invoke(common.clj:433)"]
```

If later on we execute some other queries we see that most of the threads are busy.
```
08-10 22:12:24 DEBUG metabase.middleware :: POST /api/card/264/query 200 (2 s) (15 DB calls). Jetty threads: 8/50 (50 busy, 0 idle, 3 queued)
08-10 22:12:24 DEBUG metabase.middleware :: GET /api/collection/25 200 (17 ms) (3 DB calls). Jetty threads: 8/50 (50 busy, 0 idle, 2 queued)
08-10 22:12:25 DEBUG metabase.middleware :: POST /api/card/270/query 200 (2 s) (15 DB calls). Jetty threads: 8/50 (49 busy, 1 idle, 0 queued)
08-10 22:12:25 DEBUG metabase.middleware :: POST /api/card/273/query 200 (2 s) (15 DB calls). Jetty threads: 8/50 (48 busy, 2 idle, 0 queued)
08-10 22:12:25 DEBUG metabase.middleware :: POST /api/card/266/query 200 (2 s) (15 DB calls). Jetty threads: 8/50 (47 busy, 3 idle, 0 queued)
08-10 22:12:25 DEBUG metabase.middleware :: POST /api/card/265/query 200 (2 s) (15 DB calls). Jetty threads: 8/50 (46 busy, 4 idle, 0 queued
```

We profiled the application and realized that the thread count keep growing, here is an snapshot.

![image](https://user-images.githubusercontent.com/1284879/43983854-80d19952-9cc2-11e8-8475-3d7983067cef.png)

**Comment by ikks** [comment]
Thanks for your analysis, in the organization where I work, Metabase started freezing, but with the new upgrade with x-rays we have the problem of the time.  Nice that you have profiled the issue.  

+1

**Comment by flamber** [comment]
This was most likely fixed in 0.35 with the streaming connection handling. Please comment with details if this is still a problem.
